src/gui/newwizard/pages/serverurlsetupwizardpage.cpp

```diff
@@ -17,20 +17,21 @@ ServerUrlSetupWizardPage::ServerUrlSetupWizardPage(const QUrl &serverUrl)
 
     _ui->urlLineEdit->setText(serverUrl.toString());
 
-    // first, we declare this object as an event filter
-    // this then allows us to make the controller the event filter of the current page
-    _ui->urlLineEdit->installEventFilter(this);
-
     connect(this, &AbstractSetupWizardPage::pageDisplayed, this, [this]() {
         _ui->urlLineEdit->setFocus();
     });
 
     _ui->logoLabel->setText(QString());
     _ui->logoLabel->setPixmap(Theme::instance()->wizardHeaderLogo().pixmap(200, 200));
 
-    // we already use a placeholder, but it may be overwritten by the theme
     if (!Theme::instance()->wizardUrlHint().isEmpty()) {
-        _ui->urlLineEdit->setPlaceholderText(Theme::instance()->wizardUrlHint());
+        _ui->connectTotoolButton->setText(tr("Connect to %1").arg(Theme::instance()->wizardUrlHint()));
+        connect(_ui->connectTotoolButton, &QToolButton::clicked, this, [this] {
+            _ui->urlLineEdit->setText(Theme::instance()->wizardUrlHint());
+            // TODO: trigger next...
```

Comment:
As far as I can tell there is no way I can trigger this?

Reply:
The pages are not in charge of this. At the moment, the controlling objects (the corresponding state and/or the controller) poll from the page whether the inputs are valid for instance. If you need such signalling, I guess the cleanest approach is to add a signal to the page, and connect it in the state. So far, however, this kind of functionality has not been needed.

I think UX wise, it would be better to have the button should just copy the value into the line edit, and have the user click `Next` themselves.